### PR TITLE
Add rough plan for scaffolding out component source code

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR just adds a docs page for the steps required to scaffold out a new IDS WebComponent from scratch.

**Related github/jira issue (required)**:
N/A

**Steps necessary to review your pull request (required)**:
Take a look at `doc/COMPONENTS.md`, follow the steps, and make sure the new component you make builds properly.

### CODE REVIEW

File: doc/COMPONENTS.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+- [ ] Create a folder `/src/ids-[component]`, which will contain all your new component source code.
+- [ ] Add an `ids-[component].js`, which is the WebComponent interaction code.
+- [ ] Add an `ids-[component].d.ts` for this WebComponent's TypeScript defs.
+- [ ] Add an `ids-[component].scss`, which holds all scoped styles for this WebComponent.
+- [ ] Add a `README.md` for documentation, specificiation, etc.
+
+Some important steps here include:
+
+- If this new code is an IDS Component, ensure that it imports `src/ids-base/ids-element.js` extends the `IdsElement` base component.
+- Ensure that your styles are imported in `ids-[component].js` and added to the component via the `@scss` decorator.
```
Comment (tmcconechy) [inline doc/COMPONENTS.md:16]:
```suggestion
- Ensure that your styles are imported in `ids-[component].scss` and added to the component via the `@scss` decorator.
```

File: doc/COMPONENTS.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+new HTMLWebpackPlugin({
+  template: './app/ids-[component]/index.html',
+  inject: 'body',
+  filename: 'ids-[component]/index.html',
+  title: 'New IDS Component',
+  chunks: ['ids-[component]/ids-[component]']
+}),
+```
+
+After these steps, if Webpack Dev Server is still running, make sure to restart.
```
Comment (tmcconechy) [inline doc/COMPONENTS.md:67]:
```suggestion
After these steps, if Webpack Dev Server is still running, make sure to restart. Or if you run the server with `npm run start:watch` it should restart as you save the web pack config file. Check the console to be sure it was restarted.
```

File: doc/COMPONENTS.md (hunk 1 above)
Comment (EdwardCoyle) [reply doc/COMPONENTS.md:16]: They would be doing the import in the JS file, no?

Comment (tmcconechy) [reply doc/COMPONENTS.md:16]: My mistake. I thought this was around doing this https://github.com/infor-design/enterprise-wc/blob/master/src/ids-tag/ids-tag.js#L9

**Pull request merged**